bot/on_message/bots/openai_bot.py

```diff
@@ -112,63 +128,59 @@ def should_query_weezerpedia_api(self, last_three_messages):
             print(f"An error occurred: {e}")
             return None
 
-    def fetch_openai_completion(self, message, system, incoming_message_text):
 
-        system_message = {"role": "system", "content": system}
+    def fetch_openai_completion(self, prompt_params: PromptParams, num_messages_lookback: int):
+        system_message = {"role": "system", "content": prompt_params.system_prompt}
 
-        if message.channel.id not in self.openai_sessions:
-            self.openai_sessions[message.channel.id] = []
+        if prompt_params.channel_id not in self.openai_sessions:
+            self.openai_sessions[prompt_params.channel_id] = []
 
-        messages_in_this_channel = self.openai_sessions[message.channel.id]
+        messages_in_this_channel = self.openai_sessions[prompt_params.channel_id]
 
         # Remove any existing system messages
         messages_in_this_channel = [
-            msg for msg in messages_in_this_channel if msg['role'] != 'system']
-
-        # Update the session with messages excluding old system messages
-        self.openai_sessions[message.channel.id] = messages_in_this_channel
+            msg for msg in messages_in_this_channel if msg["role"] != "system" or "[INTERNAL]" not in msg["content"]]
 
         # Add the new system message at the beginning
         new_content = [system_message] + messages_in_this_channel
 
-        # # Step 2: Add any context from Weezerpedia API if needed
+        # Replace the channel messages with the cleaned up content
+        self.openai_sessions[prompt_params.channel_id] = new_content
+
+        # Add any context from Weezerpedia API if needed
         # if weezerpedia_context := self.get_weezerpedia_context(
-        #     incoming_message_text, messages_in_this_channel
+        #     prompt_params.user_prompt, messages_in_this_channel
         # ):
         #     new_content.append(weezerpedia_context)
 
-        # Step 3: Append the user's message to the session
+        # Append the user's message to the session
         new_content.append(
-            {"role": "user", "content": f"{message.author.nick}: {incoming_message_text}"})
+            {"role": "user", "content": f"{prompt_params.nick}: {prompt_params.user_prompt}"})
 
-        # Step 4: Append any attachments to the user's message
-        new_content = self.append_any_attachments(message, new_content)
+        # Append any attachments to the user's message
+        self.append_any_attachments(prompt_params.attachment_urls, new_content)
 
-        # Step 5: Limit the number of messages in the session to 12
-        if len(messages_in_this_channel) > 12:
-            messages_in_this_channel = messages_in_this_channel[-12:]
-
-        # Step 6: Append all the new content to list of messages in this channel
-        messages_in_this_channel.extend(new_content)
```

Comment:
is this a bug? `new_content` already includes `messages_in_this_channel`. why did you add it again?